packages/next/server/web/next-url.ts

```diff
@@ -1,66 +1,67 @@
 import type { PathLocale } from '../../shared/lib/i18n/normalize-locale-path'
 import type { DomainLocale, I18NConfig } from '../config-shared'
 import { getLocaleMetadata } from '../../shared/lib/i18n/get-locale-metadata'
-import cookie from 'next/dist/compiled/cookie'
 import { replaceBasePath } from '../router'
-
-/**
- * TODO
- *
- * - Add comments to the URLNext API.
- * - Move internals to be using symbols for its shape.
- * - Make sure logging does not show any implementation details.
- * - Include in the event payload the nextJS configuration
- */
+import cookie from 'next/dist/compiled/cookie'
 
 interface Options {
+  base?: string | URL
   basePath?: string
   headers?: { [key: string]: string | string[] | undefined }
   i18n?: I18NConfig | null
   trailingSlash?: boolean
 }
 
-const REGEX_LOCALHOST_HOSTNAME =
-  /(?!^https?:\/\/)(127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}|::1)/
-
 export class NextURL extends URL {
-  private _basePath: string
-  private _locale?: {
+  #url: URL
+  #options: Options
+  #basePath: string
+  #locale?: {
```

Comment:
This syntax doesn't seem to be transpiled correctly and breaks node 12 on Azure:

```sh
 D:\a\1\s\node_modules\next\dist\server\web\next-url.js:36
         #analyzeUrl() {
                    ^

    SyntaxError: Unexpected token '('
```

x-ref: https://dev.azure.com/nextjs/next.js/_build/results?buildId=31424&view=logs&jobId=8af7cf9c-43a1-584d-6f5c-57bad8880974&j=8af7cf9c-43a1-584d-6f5c-57bad8880974&t=c6df12fd-6020-5e41-5b45-429ae6a49942

Reply:
Interesting, I will migrate it to use symbols in order to hide the internals 🙏